processor/src/main/java/org/mapstruct/ap/internal/util/Message.java

```diff
@@ -114,6 +114,9 @@
     ENUMMAPPING_NO_ELEMENTS( "'nameTransformationStrategy', 'configuration' and 'unexpectedValueMappingException' are undefined in @EnumMapping, define at least one of them." ),
     ENUMMAPPING_ILLEGAL_TRANSFORMATION( "Illegal transformation for '%s' EnumTransformationStrategy. Error: '%s'." ),
 
+    SUBCLASSMAPPING_ILLEGAL_SUBCLASS("Class '%s' is not a subclass of '%s'."),
+    SUBCLASSMAPPING_METHOD_SIGNATURE_NOT_SUPPORTED("SubClassMapping annotation can not be used on Suppliers and Runnables."),
```

Comment:
I do not think that this error message is clear. Having `void` return type means that the user is using update mappings, most likely we do not support that for now, so I would change the message to something like that.

By the way, you can do update mappings with a return type e.g. `Target update(@MappingTarget Target target, Source source)`

Reply:
So something like: `@SubClassMapping do not support update mappings.`?
Open for suggestions to make this message better, because the one above doesn't feel quite right either.